test/NuGet.Core.Tests/NuGet.Configuration.Test/SearchTreeTests.cs

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Linq;
+using FluentAssertions;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Configuration.Test
+{
+    public class SearchTreeTests
+    {
+        [Fact]
+        public void SearchTree_WithOneSource()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
```

Comment:
I don't think you need to use configuration files here. 

You should create a PackageNamespaceConfiguration indepently. These tests don't need to touch the disk. 

You'll be able to use a theory to add various tests. Right now, I'm not sure 4 tests are complete.

Reply:
Can I refactor tests in this file and add more after I get approval first?
Otherwise every time API changes I have to fix things here, it's quite time consuming.

Reply:
Can you verify whether they are polluting your standard global packages folder?

I want to make sure we don't check in something that messes with people's machines. 

I'm ok scoping down the changes so it can be reviewed faster, but keep in mind this PR is fixing 2 issues.

Reply:
I just debugged this test, this one creates new test folder then deletes(cleanup) at end up test. 
This one only creates `nuget.config` file, I would be surprised if it creates anything global packages folder.
I copied this part from below

https://github.com/NuGet/NuGet.Client/blob/b0ef730655ac770ad0e191c75698e5934411b158/test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageNamespacesConfigurationTests.cs#L21-L29

Reply:
Also I found quite more work need to do for this tests work without `nuget.config` file. Because PackageNamespacesConfiguration currently only accept `ISettings` type input for creating instance, but `ISettings` type input currently only can be created with file on disk. Since current implementation is not polluting global configuration better to keep it as it for now.

Reply:
If the search tree accepts `Dictionary<string, IReadOnlyList<string>> namespaces` type then we can eliminate the dependency on settings in the unit tests.

Reply:
I'll wait for now since it's just unit test.
1. I'm still waiting on others input on https://github.com/NuGet/NuGet.Client/pull/4039#discussion_r651304501
If we go ahead with sourceUri then we need more of nuget.config, because we need mapping between sourcekey and sourceUri from nuget.config in addition to `PackageNamespacesConfiguration` data.
2. Creating manually `Dictionary<string, IReadOnlyList<string>> namespaces` would be much more manually than editing simple text nuget.config file.

Reply:
I created follow up issue for above point 1.
Point 2 still holds for now.

Reply:
> I'll wait for now since it's just unit test

IMHO, File I/O operations are going to increase CI time. We can eliminate the calls to create and load setting file if we can modify the constructor to accept dictionary object instead.

Reply:
We need test fixture for this for [Theory] to work. I'll take a look into it but I believe it doesn't block PR.